packages/graphql-yoga/src/plugins/plugins.test.ts

```
@@ -54,4 +55,100 @@ describe('Yoga Plugins', () => {
       ],
     });
   });
+
+  describe('onValidate', () => {
+    it('should mask errors', async () => {
+      const onValidatePlugin: Plugin = {
+        onValidate({ setResult }) {
+          setResult([
+            createGraphQLError('UNMASKED ERROR', { extensions: { code: 'VALIDATION_ERROR' } }),
```

`GraphQLError` instances are not masked. The errors are masked only if it is an error is not an instance of `GraphQLError`.
And the result of `validate` function is `GraphQLError[]` so it is expected to return an array of `GraphQLError`. Yoga doesn't expect `validate` to return non GraphQL errors so they are not sent to `maskError` function.

---

Thanks @ardatan! We realize that the default[ maskError function](https://github.com/dotansimha/graphql-yoga/blob/1aa989ccdfe16a03816e432f2e6252dc84db5364/packages/graphql-yoga/src/utils/mask-error.ts#L10-L14) does not act on GraphQLErrors at all, but we _want_ to mask GraphQLErrors as well.

However, even if we write a custom `maskError` function, it is not triggered at all for GraphQLErrors.

Would it be better to use `onResultProcess` to update the result with the masked error?

---

You can use `onResultProcess` to modify the final response, yes.